frontend/pages/create-edit-user.js

```diff
@@ -0,0 +1,424 @@
+import { useState, useEffect, useRef, useReducer } from 'react';
+import {
+    client as lensClient,
+    createProfile,
+    getProfileByAddress,
+    getProfileByHandle,
+    createSetProfileWithMetadata,
+} from '../clientApi';
+import connectContract from '../utils/connectContract';
+// Components
+import Form from '../components/UI/Form';
+import Input from '../components/UI/Input';
+
+const defaultFormValues = {
+        name: '',
+        nameIsValid: null,
+        email: '',
+        emailIsValid: null,
+        username: '',
+        usernameIsValid: null,
+        website: '',
+        websiteIsValid: null,
+        bio: '',
+        bioIsValid: null,
```

Comment:
If the user has a lens profile, I think some of this data could be be already stored ( user could have set some of this data using lenster etc. ). From the looks of it in lenster repo they use the `profile` graphql query to load up the current user info ...
https://github.com/lensterxyz/lenster/blob/main/packages/lens/documents/queries/ProfileSettings.graphql

And then they default the form with the values...
https://github.com/lensterxyz/lenster/blob/50c02a2cfa5f5a45569f6e2707439aac895b90d5/apps/web/src/components/Settings/Profile/Profile.tsx#L154-L158

I don't think we have to use or display all the current user profile metadata attributes but we should should at least maintain the ones they have already set and set our own custom ones.

Reply:
Great call!

Reply:
Now data from lens fills the form if exists.
Committed at 7b3c931